### DISCUSSION TITLE
How to limit viewport detection to only once

### DISCUSSION THREAD

**Discussion by DevExplotel** [k3nsei/ng-in-viewport#1428] [Q&A] [answered]
Hi,
In my context, I want the directive to notify the detection of an element in the viewport just once so that an already detected element is no longer taken into account in future scrolls. I've looked at the guideline but don't see any such option. Please enlighten me in case it is possible in the current version otherwise I ask you to take this into consideration for future versions.

**Comment by k3nsei** [comment, marked as answer]
You can see how it's done in lazy load images example.

For example you can create directive and inject `InViewportDirective` using DI.
https://github.com/k3nsei/ng-in-viewport/blob/develop/projects/example/src/app/page-lazy-images/lazy-image.directive.ts#L39

Then just subscribe for `inViewportAction`.
https://github.com/k3nsei/ng-in-viewport/blob/develop/projects/example/src/app/page-lazy-images/lazy-image.directive.ts#L58

This is only one way how this could be achieved. There are many others.

**Reply by DevExplotel** [reply]
Fine, Thanks!
